sdks/python/apache_beam/examples/snippets/transforms/element_wise/flat_map.py

```diff
@@ -0,0 +1,220 @@
+  # [START flat_map_function]
+  import apache_beam as beam
+
+  def split_words(text):
+    return text.split(',')
+
+  with beam.Pipeline() as pipeline:
+    plants = (
+        pipeline
+        | 'Gardening plants' >> beam.Create([
+            '🍓Strawberry,🥕Carrot,🍆Eggplant',
+            '🍅Tomato,🥔Potato',
+        ])
+        | 'Split words' >> beam.FlatMap(split_words)
+        | beam.Map(print)
+    )
+    # [END flat_map_function]
+    if test:
+      test(plants)
+
+
+def flat_map_lambda(test=None):
+  # [START flat_map_lambda]
+  import apache_beam as beam
+
+  with beam.Pipeline() as pipeline:
+    plants = (
+        pipeline
+        | 'Gardening plants' >> beam.Create([
+            ['🍓Strawberry', '🥕Carrot', '🍆Eggplant'],
+            ['🍅Tomato', '🥔Potato'],
+        ])
+        | 'Flatten lists' >> beam.FlatMap(lambda elements: elements)
+        | beam.Map(print)
+    )
+    # [END flat_map_lambda]
+    if test:
+      test(plants)
+
+
+def flat_map_generator(test=None):
+  # [START flat_map_generator]
+  import apache_beam as beam
+
+  def list_elements(elements):
+    for element in elements:
+      yield element
+
+  with beam.Pipeline() as pipeline:
+    plants = (
+        pipeline
+        | 'Gardening plants' >> beam.Create([
+            ['🍓Strawberry', '🥕Carrot', '🍆Eggplant'],
+            ['🍅Tomato', '🥔Potato'],
+        ])
+        | 'Flatten lists' >> beam.FlatMap(list_elements)
+        | beam.Map(print)
+    )
+    # [END flat_map_generator]
+    if test:
+      test(plants)
+
+
+def flat_map_multiple_arguments(test=None):
+  # [START flat_map_multiple_arguments]
+  import apache_beam as beam
+
+  def split_words(text, delimiter=None):
+    return text.split(delimiter)
+
+  with beam.Pipeline() as pipeline:
+    plants = (
+        pipeline
+        | 'Gardening plants' >> beam.Create([
+            '🍓Strawberry,🥕Carrot,🍆Eggplant',
+            '🍅Tomato,🥔Potato',
+        ])
+        | 'Split words' >> beam.FlatMap(split_words, delimiter=',')
+        | beam.Map(print)
+    )
+    # [END flat_map_multiple_arguments]
+    if test:
+      test(plants)
+
+
+def flat_map_side_inputs_singleton(test=None):
+  # [START flat_map_side_inputs_singleton]
+  import apache_beam as beam
+
+  with beam.Pipeline() as pipeline:
+    delimiter = pipeline | 'Create delimiter' >> beam.Create([','])
+
+    plants = (
+        pipeline
+        | 'Gardening plants' >> beam.Create([
+            '🍓Strawberry,🥕Carrot,🍆Eggplant',
+            '🍅Tomato,🥔Potato',
+        ])
+        | 'Split words' >> beam.FlatMap(
+            lambda text, delimiter: text.split(delimiter),
+            delimiter=beam.pvalue.AsSingleton(delimiter),
+        )
+        | beam.Map(print)
+    )
+    # [END flat_map_side_inputs_singleton]
+    if test:
+      test(plants)
+
+
+def flat_map_side_inputs_iter(test=None):
+  # [START flat_map_side_inputs_iter]
+  import apache_beam as beam
+
+  def validate_durations(plant, valid_durations):
```

Comment:
I'd prefer a unique example transform for `flat_map_side_inputs_iter` that could not be replaced by a filter

Reply:
Got it, I added an "extra" step of normalizing it to lowercase, so it's now slightly smarter than a regular filter.